### DISCUSSION TITLE
Make it independent of network library (okhttp)

### DISCUSSION THREAD

**Discussion by rfc2822** [bitfireAT/dav4jvm#55] [Ideas]
Use cases:

- anyone on JVM/Kotlin who wants a WebDAV library, but already uses something else than okhttp, could use it then
- we could use [Nextcloud Android-SingleSignOn](https://github.com/nextcloud/Android-SingleSignOn) for DAVx5 (it has to be used on network level, instead of okhttp).

**Comment by ArnyminerZ** [comment]
Well, I've been taking a look, and this would be pretty much rewriting the whole library 😅 

But I think it would be really good. It would give support to ktor, for example, which would be compatible with Kotlin Multiplatform.

_okhttp is technically compatible, but I'd say that not on all platforms, and also not with full support so..._

**Comment by ArnyminerZ** [comment]
Also just a quick note, the example at Nextcloud's README is not up to date, they have new methods, at least at the source code. I don't know if they have been released or not, but the example only shows a single stream for output, which would not allow headers to be obtained from the request for example. It's still a pretty basic implementation what they have though. It would be really cool if we could just extract the tokens authorized and call it a day to be honest.

**Edit:** They do some AIDL magic to communicate with the Nextcloud app to extract token. Maybe we can just copy some of it if we want to implement Nextcloud SSO in DAVx⁵. I wouldn't love it though. AIDL is pretty annoying to work with...

**Edit 2:** ~It's actually not, I _think_ the token is stored directly, and can be obtained by [`getAccountToken`](https://github.com/nextcloud/Android-SingleSignOn/blob/main/lib/src/main/java/com/nextcloud/android/sso/api/NetworkRequest.java#L70), which should be pretty easy to obtain from `SingleAccountHelper.getCurrentSingleSignOnAccount`~
